[geometry.rst] tested and updated geometry.rst of the pyqgis cookbook (#3522)
qgis/QGIS-Documentation: source/docs/pyqgis_developer_cookbook/geometry.rst

    gPnt = QgsGeometry.fromPointXY(QgsPointXY(1,1))
    gLine = QgsGeometry.fromPolyline([QgsPoint(1, 1), QgsPoint(2, 2)])

>     gPolygon = QgsGeometry.fromPolygonXY([[QgsPointXY(1, 1), QgsPointXY(2, 2), QgsPointXY(2, 1)]])

Author: Thanks, fixed in [afb82e7d106a7a2d019f6a673fdc2874a01e628d](https://github.com/qgis/QGIS-Documentation/pull/3522/commits/afb82e7d106a7a2d019f6a673fdc2874a01e628d)

  >>> gPnt.asPoint()
  <QgsPointXY: POINT(1 1)>

>   >>> gPnt.asPoint()

Reviewer: could you please remove the python `>>>` strings?

Author: Thank you. I now removed the interpreter prompt strings and put the expected outputs in comments. I also put he qLine.asPolyline() example back, that was not meant to be removed.

properties of geometries, such as area (in the case of polygons) or lengths
(for polygons and lines)

Let's see an example that combines iterating over the features in a
given layer and performing some geometric computations based on their

> geometries. The below code will caompute and print the area and perimeter of each country in the 'countries' layer within our tutorial QGIS project.

Reviewer 2: typo. Also favour shorter lines, easier to read/review
```suggestion
geometries. The below code will compute and print the area and perimeter of
each country in the :guilabel:`countries` layer within our tutorial QGIS project.
```
Sorry, I did not follow discussion so, would the "tutorial QGIS project" data be available to users? Would that mention mean anything to them?

Author: Yes we discusseed on the hackfest. But I am not aware how / where it would be available from. Would have wanted to put a link with the url instead of just mentioning it. It would have to be updated once the location of the demo QGIS project file with the demo data (.gpkg) is known

methods from the :class:`QgsGeometry <qgis.core.QgsGeometry>` class. For a more powerful area and
distance calculation, the :class:`QgsDistanceArea <qgis.core.QgsDistanceArea>` class can be used, which can perform ellipsoid based calculations:

.. code-block:: python

  d = QgsDistanceArea()
  d.setEllipsoid('WGS84')

  layer = QgsProject.instance().mapLayersByName('countries')[0]

  # let's filter for countries that begin with Z, then get their features
  query = '"name" LIKE \'Z%\''
  features = layer.getFeatures(QgsFeatureRequest().setFilterExpression(query))

  for f in features:
    geom = f.geometry()
    name = f.attribute('NAME')
    print(name)
    print("Perimeter (m):", d.measurePerimeter(geom))
    print("Area (m2):", d.measureArea(geom))
    
    # let's calculate and print the area again, but this time in square kilometers
    print("Area (km2):", d.convertAreaMeasurement(d.measureArea(geom), QgsUnitTypes.AreaSquareKilometers))

Alternatively, you may want to know the distance and bearing between two points.

.. code-block:: python

  d = QgsDistanceArea()
  d.setEllipsoid('WGS84')

>   # A let's create two points. 

Reviewer 2: ```suggestion
  # Let's create two points. 
```
?

Author: oops. Sorry for that - looks like my pre-commit hook did not run, i will fix that and pay close attention next time

Author: FYI it is my first pull request to QGIS and a newcomer to the contributor side of QGIS, I try to do better on my next PR :)

Reviewer 2: There's no problem @arongergely We all make mistakes (if we can call this a mistake) and this is why we review each other PRs. Btw for a first contribution, it's rather good. Thanks.

>   # Santa claus is a workoholic and needs a summer break, 

Reviewer 3: ```suggestion
  # Santa claus is a workaholic and needs a summer break, 
```

Reviewer 2: Does that guy not work only one day in the year? OK 24h non stop but still... workaholic?

Author: hm. fair point - i just wanted something funny- I think he would work all year and collating all the wishes, meanwhile monitoring all kid's behaviour , etc. Isn't that the "santa" protocol?

Reviewer 2: > i just wanted something funny

It made me laugh 😄
